src/main/java/org/commonjava/indy/service/repository/controller/AdminController.java

```diff
@@ -188,11 +195,10 @@ public void delete( final StoreKey key, final String user, final String changelo
             if ( store != null && deleteContent )
             {
                 logger.info( "Delete content of {}", key );
-                //TODO: here should send event to notify store service to delete file contents for this repo
-                //                deleteContent( store );
+                purgeFilesystem( key.toString() );
```

Comment:
The logic looks good to me. But one concern is the performance. I'm not sure if this purge operation is a huge time-consumed. If so I'd recommend to run it in async mode. (threaded or Quarkus reactive mode)

Reply:
sounds good. I will push another pr to make is async.